### ISSUE TITLE
USWDS - Feature: CSS logical properties for multilingual support

### ISSUE THREAD

**Issue by mejiaj** [uswds/uswds#5336]
### Is your feature request related to a problem? Please describe.

This new native CSS feature improves support for different writing modes in different layouts (like RTL).

Currently margin-left and margin-right could potentially break designs for RTL or vertical text layouts.

### Describe the solution you'd like

CSS logical properties for components and utilities. 

### Describe alternatives you've considered

A new theme setting that allows users to enable/disable logical properties.

### Additional context

<img src="https://github.com/uswds/uswds/assets/3385219/3a366b37-5ba8-46d3-a752-9bd788f027ac" width="250" />

**Resources**
- [MDN docs on logical properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_logical_properties_and_values)
- [Example of `margin-block-end`](https://developer.mozilla.org/en-US/docs/Web/CSS/margin-block-end)

### Next steps

- [x] ~Identify core properties based on impact, LOE, and browser support.~ Identified `margin, padding, position`.
- [ ] Identify areas of codebase that could make the most impact with minimal changes. 
_For example: functions, mixins, and placeholders vs component or layout styles._ 
    - [x] #6030
    - [ ] Add logical properties for Identifier
- [ ] Confirm is new theme settings flag to enable logical vs traditional is needed.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Feature+Request%22) for duplicate feature requests.

**Comment by mgifford** [comment]
I just wanted to chime in on this issue after hearing about it at Drupal GovCon. Moving from a left / right (spatial) framing to a first / last (logical) would be really great.  

It would help bidirectional sites (bidi) sites a lot. Any site that requires English and either Arabic, Farsi (Persian) or Hebrew would benefit from this. 

It would also help for building a more mobile friendly approach. What does left/right mean in the context of a mobile site? 

Furthermore, as we move to more aural interfaces, it would help shift our awareness about how content is displayed in linear aural interfaces. Obviously screen readers are a type of aural interface, but so are Alexa, Siri and Google Home. 

Finally, this shift will take a long while to go through legacy content using older LTR models.  We should have a transitional window for this, but ultimately, need to make a logical ordering implemented earlier, so that it will cost less to transition in the future.

**Comment by mejiaj** [comment]
@mgifford thanks for that note. Left to right makes sense and it'd be nice to move this forward.

I've created an issue in #6030 to explore ways to address.

**Comment by mejiaj** [comment]
Also exploring building with this approach in web components, specifically uswds/web-components#39.

**Comment by ryparker** [comment]
We're keeping this open to try logical properties in the Identifier component first. This could improve layouts for languages written right to left, such as Arabic and Hebrew.

The next step is a working right-to-left example. We also need to check that existing left-to-right layouts, left/right utility classes, and custom site styles keep working unchanged.

What we learn from the Identifier can help us decide whether to extend this to other components or add a theme setting.
